### PR TITLE
CORE-255 - Viewing Membership Feedback

### PR DESCRIPTION
**Pull request by AxonC**
This is beautiful by the way. (Travis won't pass due to an ongoing issue)

### CODE REVIEW

**Comment by AxonC** [comment]
Required #1391 before merging.

File: app/Nova/Actions/SendFeedback.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        /** @var \App\Models\Mship\Feedback\Feedback */
+        $feedback = $models->first();
+        $actioner = auth()->user();
+
+        if ($feedback->sent_at != null) {
+            return Action::danger('This feedback is already sent to the user.');
```
Comment (CalumTowers) [inline app/Nova/Actions/SendFeedback.php]: "This feedback **has** already **been** sent to the **member**."

File: app/Nova/Actions/ActionFeedback.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        /** @var \App\Models\Mship\Feedback\Feedback */
+        $feedback = $models->first();
+        $actioner = auth()->user();
+
+        if ($feedback->actioned_at != null) {
+            return Action::danger('This feedback is already actioned!');
```
Comment (CalumTowers) [inline app/Nova/Actions/ActionFeedback.php]: "This feedback **has** already **been** actioned."

File: app/Nova/Actions/ActionFeedback.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+     *
+     * @param  \Laravel\Nova\Fields\ActionFields  $fields
+     * @param  \Illuminate\Support\Collection  $models
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        /** @var \App\Models\Mship\Feedback\Feedback */
+        $feedback = $models->first();
+        $actioner = auth()->user();
```
Comment (CalumTowers) [inline app/Nova/Actions/ActionFeedback.php]: I would inline the `$actioner` variable to try and tidy up a little?

File: app/Nova/Actions/ActionFeedback.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * Get the fields available on the action.
+     *
+     * @return array
+     */
+    public function fields()
+    {
+        return [
+            Textarea::make('Comment')->rules('required', 'min:3'),
```
Comment (CalumTowers) [inline app/Nova/Actions/ActionFeedback.php]: We probably want a higher minimum than 3 characters for a comment.

File: app/Nova/Actions/SendFeedback.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     *
+     * @param  \Laravel\Nova\Fields\ActionFields  $fields
+     * @param  \Illuminate\Support\Collection  $models
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        /** @var \App\Models\Mship\Feedback\Feedback */
+        $feedback = $models->first();
+        $actioner = auth()->user();
```
Comment (CalumTowers) [inline app/Nova/Actions/SendFeedback.php]: Inline here too.

File: app/Nova/Actions/SendFeedback.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+        if ($feedback->sent_at != null) {
```
Comment (CalumTowers) [inline app/Nova/Actions/SendFeedback.php]: Can this be simplified to `if ($feedback->sent_at)`?

File: app/Nova/Actions/ActionFeedback.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+     * @param  \Illuminate\Support\Collection  $models
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        /** @var \App\Models\Mship\Feedback\Feedback */
+        $feedback = $models->first();
+        $actioner = auth()->user();
+
+        if ($feedback->actioned_at != null) {
```
Comment (CalumTowers) [inline app/Nova/Actions/ActionFeedback.php]: Can this be simplified to `if ($feedback->actioned_at)`?

**Pull request merged**